src/josepy/json_util.py

```diff
@@ -409,17 +412,17 @@ def decode_csr(b64der):
 class TypedJSONObjectWithFields(JSONObjectWithFields):
     """JSON object with type."""
 
-    typ = NotImplemented
+    typ = NotImplemented  # type: Any
```

Comment:
Reading over https://github.com/python/mypy/issues/4791, it seems like the best choice here would be:

```suggestion
    typ: str = NotImplemented 
```